base/math.jl

```diff
@@ -696,7 +696,20 @@ function _hypot(x, y)
     end
     return h*scale*oneunit(axu)
 end
-_hypot(x::Float16, y::Float16) = Float16(_hypot(Float32(x), Float32(y)))
+@inline function _hypot(x::Float32, y::Float32)
+    if isinf(x) || isinf(y)
+        return Inf32
+    end
```

Comment:
Why is this special case needed?   The `muladd` code below will give you infinity anyway in this case.

Reply:
https://github.com/JuliaLang/julia/pull/42122#discussion_r702982046